### PR TITLE
Improve testing and coverage of glm second level

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Following #2764 this PR proposes to improve testing and coverage for `nilearn/glm/second_level/second_level.py`.
Helper functions from this file are also unit-tested in a consistent way.

### CODE REVIEW

File: nilearn/glm/tests/test_second_level.py [hunk 1]
```diff
@@ -32,6 +34,161 @@
...
+def test_check_second_level_input():
+    from nilearn.glm.second_level.second_level import _check_second_level_input
+    with pytest.raises(ValueError,
+                       match="A second level model requires a list with at "
+                             "least two first level models or niimgs"):
+        _check_second_level_input([FirstLevelModel()], pd.DataFrame())
+    with pytest.raises(ValueError,
+                       match="Model sub_1 at index 0 has not been fit yet"):
+        _check_second_level_input([FirstLevelModel(subject_label="sub_{}".format(i))
+                                   for i in range(1,3)], pd.DataFrame())
```
Comment (bthirion) [inline nilearn/glm/tests/test_second_level.py:46]:
```suggestion
                                   for i in range(1, 3)], pd.DataFrame())
```

**Review by bthirion — COMMENTED** [review]
Looks good ! Thx,

File: nilearn/glm/tests/test_second_level.py [hunk 2]
```diff
@@ -32,6 +34,159 @@
...
+    second_level_input = pd.DataFrame({'map_name': ["a", "b"] ,
+                                        'effects_map_path': [fmri_data[0], "bar"]})
+    assert _infer_effect_maps(second_level_input, "a") == [fmri_data[0]]
+    with pytest.raises(ValueError,
+                       match="File not found: 'bar'"):
+        _infer_effect_maps(second_level_input, "b")
+    assert _infer_effect_maps([fmri_data[0]], None) == [fmri_data[0]]
+    contrast = np.eye(rk)[1]
+    second_level_input = [FirstLevelModel(mask_img=mask)] * 2
+    for i,model in enumerate(second_level_input):
```
Comment (bthirion) [inline nilearn/glm/tests/test_second_level.py:181]:
```suggestion
    for i, model in enumerate(second_level_input):
```

**Review by bthirion — COMMENTED** [review]
Last micro-glitch and it's good for merge.

File: nilearn/glm/tests/test_second_level.py (hunk 2 above)
Comment (NicolasGensollen) [reply nilearn/glm/tests/test_second_level.py]: Thanks!

**Pull request merged**